p2p/utils.go

```diff
@@ -135,12 +135,13 @@ func saveKey(priv crypto.PrivKey, fp string) (err error) {
 	return f.Close()
 }
 
-// TODO: implement LEB128 variable-length encoding
-
-// Decodes a byte array to uint64 using LEB128 variable-length encoding
-// func leb128ToUint64(in []byte) uint64 {
-// 	return leb128.ToUInt64(in)
-// }
+// encodeMessageLEB128 applies leb128 variable length encoding
+func encodeMessageLEB128(encMsg []byte) []byte {
+	n := leb128.ToUInt64(encMsg)
```

Comment:
I don't think this is correct?
`ToUint64` decodes a LEB128 encoded byte array
I think you want to use `out := leb128.FromUint64(uint64(len(encMsg))); return append(out, encMsg...)`
https://godoc.org/github.com/filecoin-project/go-leb128

Reply:
oof, that makes more sense

Reply:
fixed!